test/Microsoft.AspNetCore.Mvc.Formatters.Json.Test/JsonInputFormatterTest.cs

```diff
@@ -105,7 +150,95 @@ public class JsonInputFormatterTest
             Assert.Equal("Person Name", userModel.Name);
             Assert.Equal(30, userModel.Age);
 
-            // Reading again should not fail as the request body should have been buffered by the formatter
+            Assert.False(httpContext.Request.Body.CanSeek);
+            result = await formatter.ReadAsync(context);
+
+            // Assert
+            Assert.False(result.HasError);
+            Assert.Null(result.Model);
+        }
+
+        [Fact]
+        public async Task SuppressInputFormatterBufferingSetToTrue_UsingMvcOptions_DoesNotBufferRequestBody()
+        {
+            // Arrange
+            var content = "{name: 'Person Name', Age: '30'}";
+            var logger = GetLogger();
+            var mvcOptions = new MvcOptions();
+            mvcOptions.SuppressInputFormatterBuffering = true;
```

Comment:
Can you assign this field as part of `// Act` and say it's intentional to verify we aren't caching a value type?

Reply:
This hasn't changed yet.

Reply:
This particular line instance does not verify the mutation of options. I have it another tests where I did move this line to 'Act'